src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Quote.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Helper_Frontend_Quote extends Mage_Core_Helper_Abstract
+{
+    /**
+     * @param Mage_Customer_Model_Customer $customer
+     *
+     * @return Mage_Sales_Model_Quote
+     */
+    public function createNewCustomerQuote($customer)
+    {
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = Mage::getModel('sales/quote')->assignCustomer($customer);
```

Comment:
it's a good start, but I was talking about Shopgate_Cloudapi_Model_Api2_Carts_Rest_Customer_V2::setCustomerData() function, it assigns more than just address. See if you can pull that setter into the helper and reuse by endpoint + your code.

Reply:
setCustomerData outsourced to a new helper